eras/shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs

```diff
@@ -192,6 +193,12 @@ instance Group (MaryValue c) where
 
 instance Abelian (MaryValue c)
 
+instance CC.Crypto c => Split (MaryValue c) where
+  vsplit mv 0 = ([], coin mv)
+  vsplit (MaryValue n ma) m
+    | m <= 0 = error "must split maryvalue into positive parts"
+    | otherwise = (take (fromIntegral m) (repeat (MaryValue (n `div` m) ma)), Coin (n `rem` m))
+
```

Comment:
@lehins @Soupstraw Need your approval on the correctness of this implementation.

Reply:
Nevermind. I found an instance declared in `MaryEraGen`